### PR TITLE
DB datagrid store sync

### PR DESCRIPTION
**Pull request by goransle**

### CODE REVIEW

**Review by karolkolodziej — COMMENTED** [review]
Questions:
- Will these `todos` be covered under this PR os should we discuss that later?
- Could you please update this PR with the latest `enhancement-DataGridComponent` and test this in `dashboard-minimal` demo?

**Comment by goransle** [comment]
Removed the component TODOs that I added to Asana/fixed. Leaving the one's in DataGrid for further review

File: ts/DataGrid/DataGrid.ts [hunk 1]
```diff
@@ -339,11 +353,22 @@ class DataGrid implements DataEventEmitter<DataGrid.Event> {
         const rowCount = this.getRowCount();
 
         for (let j = 0; j < this.rowElements.length && i < rowCount; j++, i++) {
-            const cellElements = this.rowElements[j].childNodes;
+            const rowElement = this.rowElements[j];
+            rowElement.dataset.rowIndex = String(i);
```
Comment (goransle) [inline ts/DataGrid/DataGrid.ts:357]: @oysteinmoseng any a11y concerns regarding the dataset approach?

Comment (oysteinmoseng) [reply ts/DataGrid/DataGrid.ts:357]: All good!

**Comment by karolkolodziej** [comment]
I will look into that after merging https://github.com/highcharts/highcharts/pull/17987

File: ts/Extensions/DashboardPlugins/DataGridComponent.ts [hunk 2]
```diff
@@ -74,20 +71,64 @@ class DataGridComponent extends Component<DataGridComponent.ChartComponentEvents
...
+                cell &&
+                cell instanceof HTMLElement
+            ) {
+                const dataTableRowIndex = parentRow
+                    .dataset.rowIndex;
+                const { columnName } = cell.dataset;
+
+                if (
+                    dataTableRowIndex !== void 0 &&
+                                    columnName !== void 0
```
Comment (karolkolodziej) [inline ts/Extensions/DashboardPlugins/DataGridComponent.ts:98]: Some weird indentation

**Review by karolkolodziej — APPROVED** [review]

File: ts/Extensions/DashboardPlugins/DataGridComponent.ts (hunk 2 above)
Comment (goransle) [reply ts/Extensions/DashboardPlugins/DataGridComponent.ts:98]: Something off with my ESLint, but only when not using VSCode 🤔

**Pull request merged**